src/services/messageToRoomSender.js

```diff
@@ -0,0 +1,40 @@
+const { type } = require('express/lib/response')
+const { Message } = require('../entities')
+const { makeMessageGenerator } = require('../utils/messages')
+const Filter = require('bad-words')
+
+const filter = new Filter()
+
+class MessageToRoomSender {
+    constructor({ messageRepo }) {
+        this.messageRepo = messageRepo
+    }
+
+    call({ user, room, message, type }) {
+        try {
+            return this.send({ user, room, message, type })
+        } catch (error) {
+            return { error: error.message || error }
+        }
+    }
+
+    async send({ user, room, message, type }) {
+        if (filter.isProfane(message)) {
+            return { error: 'Profanity is not allowed.' }
+        }
+        const generatedMessage = makeMessageGenerator(user)(message)
```

Comment:
generate message must be injected, that will allow you yo use one message sender for location and common message creation